idaes/ui/fsvis/static/js/paper.js

```diff
@@ -63,6 +63,28 @@ export class Paper {
         // context menu
         self._paper.on("element:contextmenu", function(cellView, evt) {
             cellView.model.rotate(90)
+            // This is needed to keep the text labels for the unit models in the correct orientation
+            // x and y were specifically picked to keep the label in the same place 
+            // in relation to the unit model (bottom middle)
+            let x = 0
+            let y = 0
+            if (cellView.model.angle() == 0) {
```

Comment:
You could consider a `switch` statements with a `default` case instead of the `if...else if` chain with `let` variables. The two would be equivalent so that'd mostly a matter of style/clarity.

Reply:
Done